elytron/src/test/resources/org/wildfly/extension/elytron/tls-test.xml

```diff
@@ -23,21 +23,25 @@
     </credential-stores>
     <tls>
         <key-stores>
-            <key-store name="FireflyKeystore" type="JKS">
-                <file path="firefly.keystore" relative-to="jboss.server.config.dir"/>
+            <key-store name="FireflyKeystore" >
                 <credential-reference store="test" alias="master-password-alias"/>
+                <implementation type="JKS" />
```

Comment:
I'm a bit confused by this discussion. KeyStoreDefinition.TYPE is a required attribute. So I'm not sure how defaults or "go for the JDK default if not specified" come into it, unless the concern is because there's other work that will make that attribute not required.

FWIW, the persistence rules are we persist an attribute if the user explicitly configured it, whether the value matches the default value or not. But we don't write a default value if the user-provided value is undefined. The presence or absence of an attribute group *shouldn't* affect that.

I haven't looked at the details of the marshaller code but my expectation is that if an attribute group has no defined attributes, then no element for that group would be written.